src/coreComponents/physicsSolvers/SolverStatistics.cpp

```diff
@@ -101,59 +134,154 @@ void SolverStatistics::logTimeStepCut()
+{
+  std::vector< TableData::CellData > residualsNormCells;
+
+  auto hasValue = []( real64 value )
   {
-    if( printOuterLoopIterations )
-    {
-      logStat( "successful outer loop iterations", m_numSuccessfulOuterLoopIterations );
-    }
-    logStat( "successful nonlinear iterations", m_numSuccessfulNonlinearIterations );
-    if( printLinearIterations ) // don't print for the outer iterations in sequential schemes
-    {
-      logStat( "successful linear iterations", m_numSuccessfulLinearIterations );
-    }
+    return std::fabs( value - std::numeric_limits< real64 >::max()) > std::numeric_limits< real64 >::epsilon();
+  };
+
+  struct ResidualInfo
+  {
+    const real64 & value;
+    string_view columnName;
+  };
+  ResidualInfo residuals[] = {
+    { m_residualMass, "RMass" },
+    { m_residualVol, "RVol" },
+    { m_residualEnergy, "REnergy" },
+    { m_residualFlow, "RFlow" },
+    { m_residualBubbleDisp, "RBubbleDisp" },
+    { m_residualFracture, "RFrac" },
+    { m_residualStick, "Rstick" },
+    { m_residualSlip, "Rslip" },
+    { m_residualOpen, "Ropen" },
+    { m_residualSolid, "RSolid" },
+    { m_residualContact, "RContact" },
+    { m_residualProppant, "RProppant" },
+    { m_residualWell, "RWell" },
+    { m_residualDamage, "RDamage" },
+    { m_totalResidual, "RTotal" }
+  };
 
-    logStat( "time step cuts", m_numTimeStepCuts );
-    if( printOuterLoopIterations )
+  std::cout << "m_numTimeSteps " << m_numTimeSteps << " m_currentNewtonIter "<< m_currentNewtonIter << std::endl;
+  if( m_numTimeSteps == 0 && m_currentNewtonIter == 0 )
+  {
+    for( auto const & residual : residuals )
     {
-      logStat( "discarded outer loop iterations", m_numDiscardedOuterLoopIterations );
+      if( hasValue( residual.value ) )
+      {
+        std::cout << " residual.columnName "<< residual.columnName << std::endl;
+        m_nonLinearNormsLayout->addColumn( residual.columnName );
+      }
     }
-    logStat( "discarded nonlinear iterations", m_numDiscardedNonlinearIterations );
-    if( printLinearIterations ) // don't print for the outer iterations in sequential schemes
+  }
+
+  residualsNormCells.emplace_back( TableData::CellData( {CellType::Value,
+                                                         GEOS_FMT( "{}", m_numTimeSteps )} ));
+  residualsNormCells.emplace_back( TableData::CellData( {CellType::Value,
+                                                         GEOS_FMT( "{}", m_currentNewtonIter )} ));
+
+  for( auto const & residual : residuals )
+  {
+    if( hasValue( residual.value ) )
     {
-      logStat( "discarded linear iterations", m_numDiscardedLinearIterations );
+      residualsNormCells.emplace_back( TableData::CellData( {CellType::Value,
+                                                             GEOS_FMT( "{}", residual.value )} ));
     }
   }
+
+  m_nonLinearNormsData.addRow( residualsNormCells );
 }
+
+void ConvergenceStatistics::outputResidualNorm( bool writeCSV ) const
```

Comment:
Just to standardize a bit?
```suggestion
void ConvergenceStatistics::outputStatistics( bool writeCSV ) const
```